### ISSUE TITLE
problem for evaluating mamujoco

### ISSUE THREAD

**Issue by Lin-c-x** [zbzhu99/madiff#4]
i can run the code to train mamujoco,but when i evaluate it , it occurs some problems
<img width="479" alt="image" src="https://github.com/zbzhu99/madiff/assets/74222039/a7984de8-ca51-438b-a51d-5187c436abef">

**Comment by zbzhu99** [comment]
Sorry for the late reply. Could you share the result of executing `pip list` in the Python environment you're using?

**Comment by Lin-c-x** [comment]
absl-py                      2.1.0
aiofile                      3.8.8
aiofiles                     23.2.1
argcomplete                  3.2.2
astunparse                   1.6.3
atomicwrites                 1.4.1
attrs                        23.2.0
black                        23.1.0
boto3                        1.34.55
botocore                     1.34.55
bracex                       2.4
cachetools                   4.2.4
caio                         0.9.13
certifi                      2024.2.2
cffi                         1.16.0
charset-normalizer           3.3.2
click                        8.1.7
cloudpickle                  1.3.0
contourpy                    1.1.1
cycler                       0.12.1
Cython                       0.29.28
D4RL                         1.1
ddpg-agent                   0.0.1          /data/lcx/madiff/third_party/ddpg-agent
deepdiff                     6.7.1
diffusers                    0.24.0
dill                         0.3.8
dm-control                   1.0.8
dm-env                       1.6
dm-tree                      0.1.8
einops                       0.7.0
enum34                       1.1.10
expandvars                   0.12.0
fasteners                    0.19
filelock                     3.13.1
flatbuffers                  23.5.26
fonttools                    4.49.0
fsspec                       2024.2.0
functional-notations         0.5.2
future                       1.0.0
gast                         0.4.0
gitdb                        4.0.5
GitPython                    3.1.9
glfw                         2.7.0
google-auth                  2.28.1
google-auth-oauthlib         1.0.0
google-pasta                 0.2.0
graphviz                     0.20.1
grpcio                       1.62.0
gtimer                       1.0.0b5
gym                          0.10.8
gym-notices                  0.0.8
gymnasium                    0.27.1
gymnasium-notices            0.0.1
h5py                         3.10.0
html5tagger                  1.3.0
httptools                    0.6.1
huggingface-hub              0.21.3
idna                         3.6
imageio                      2.34.0
imageio-ffmpeg               0.4.9
importlib-metadata           7.0.1
isort                        5.12.0
jax-jumpy                    1.0.0
jaynes                       0.8.11
Jinja2                       3.1.3
jmespath                     1.0.1
joblib                       0.16.0
keras                        2.13.1
kiwisolver                   1.4.5
labmaze                      1.0.6
libclang                     16.0.6
lxml                         5.1.0
madiff                       0.0.0          /data/lcx/madiff
MAMujoco                     1.1.0
Markdown                     3.5.2
MarkupSafe                   2.1.5
matplotlib                   3.6.3
mjrl                         1.0.0
ml-logger                    0.8.69
mock                         5.1.0
more-itertools               10.2.0
mpyq                         0.2.5
mujoco                       2.3.0
mujoco-py                    2.1.2.14
multiagent                   0.0.1          /data/lcx/madiff/third_party/multiagent-particle-envs
multidict                    6.0.5
mypy-extensions              1.0.0
networkx                     3.1
numpy                        1.24.3
numpy-stl                    2.17.1
oauthlib                     3.2.2
OG-MARL                      0.0.1          /data/lcx/madiff/third_party/og-marl
opencv-python                4.9.0.80
opt-einsum                   3.1.0
ordered-set                  4.1.0
packaging                    23.0
pandas                       1.3.5
params-proto                 2.9.6
path                         15.0.0
pathspec                     0.12.1
pillow                       10.2.0
pip                          23.3.1
platformdirs                 4.2.0
pluggy                       1.4.0
portpicker                   1.6.0
protobuf                     3.20.3
psutil                       5.9.8
py                           1.11.0
pyasn1                       0.5.1
pyasn1-modules               0.3.0
pybullet                     3.2.6
pycparser                    2.21
pycurl                       7.45.3
pygame                       2.5.2
pyglet                       1.5.0
PyOpenGL                     3.1.5
pyparsing                    2.2.2
pyrsistent                   0.16.0
PySC2                        3.0.0
pytest                       3.8.2
python-dateutil              2.7.3
python-utils                 2.4.0
pytz                         2024.1
PyWavelets                   1.4.1
PyYAML                       6.0
regex                        2023.12.25
requests                     2.31.0
requests-futures             1.0.1
requests-oauthlib            1.3.1
requests-toolbelt            1.0.0
rsa                          4.9
ruamel.yaml                  0.18.6
ruamel.yaml.clib             0.2.8
s2clientprotocol             5.0.12.91115.0
s2protocol                   5.0.12.91115.0
s3transfer                   0.10.0
safetensors                  0.4.2
sanic                        23.12.1
Sanic-Cors                   2.2.0
sanic-routing                23.12.0
scikit-image                 0.19.3
scikit-video                 1.1.11
scipy                        1.10.1
seaborn                      0.13.2
setuptools                   68.2.2
six                          1.16.0
sk-video                     1.1.10
SMAC                         1.0.0
smmap                        3.0.5
subprocess32                 3.5.4
tensorboard                  2.13.0
tensorboard-data-server      0.7.2
tensorboardX                 2.0
tensorflow                   2.13.1
tensorflow-estimator         2.13.0
tensorflow-io-gcs-filesystem 0.34.0
termcolor                    2.4.0
tifffile                     2023.7.10
tomli                        2.0.1
torch                        1.12.1+cu113
torchviz                     0.0.2
tqdm                         4.66.2
tracerite                    1.1.1
typed-argument-parser        1.7.2
typing_extensions            4.5.0
typing-inspect               0.9.0
ujson                        5.9.0
urllib3                      1.26.18
uvloop                       0.19.0
waterbear                    2.6.8
wcmatch                      8.5.1
websocket-client             1.7.0
websockets                   12.0
Werkzeug                     3.0.1
wheel                        0.41.2
whichcraft                   0.6.1
wrapt                        1.16.0
zipp                         3.17.0

**Comment by Lin-c-x** [comment]
And when I run spread of the mpe environment, the result is above 500. Is there anythig need to change? I try to change the return_scale, but seems useless.

**Comment by zbzhu99** [comment]
Please try installing the local multiagent mujoco environment with:

```bash
pip install -e ./third_party/multiagent_mujoco
```

**Comment by zbzhu99** [comment]
> And when I run spread of the mpe environment, the result is above 500. Is there anythig need to change? I try to change the return_scale, but seems useless.

To be consistent with the results in the OMAR paper, we reported normalized scores on MPE. Details on normalization can be seen in Appendix E.1 of our arXiv paper.

<img width="1680" alt="截屏2024-04-09 12 31 10" src="https://github.com/zbzhu99/madiff/assets/32090492/011d9d3c-ffff-48ab-91c3-78be328561fb">

**Comment by Lin-c-x** [comment]
thank you for your reply
